File: backend/src/main/java/hu/bsstudio/raktr/dto/rent/RentCreateDto.java
package hu.bsstudio.raktr.dto.rent;

import hu.bsstudio.raktr.dal.value.RentType;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class RentCreateDto {

    @NotNull
    private RentType type;

    @NotBlank
    private String destination;

    @NotNull
    private UUID issuerId;

    @NotBlank
    private String renterName;

    @NotNull
    private LocalDate outDate;

    @NotNull
    private LocalDate expectedReturnDate;

}

File: frontend/src/app/model/rent/rentCreateDto.ts
import { RentType } from './rentType';

export class RentCreateDto {
  type: RentType;
  destination: string;
  issuerId: string;
  renterName: string;
  outDate: Date;
  expectedReturnDate: Date;

  constructor(
    type: RentType,
    destination: string,
    issuerId: string,
    renterName: string,
    outDate: Date,
    expectedReturnDate: Date,
  ) {
    this.type = type;
    this.destination = destination;
    this.issuerId = issuerId;
    this.renterName = renterName;
    this.outDate = outDate;
    this.expectedReturnDate = expectedReturnDate;
  }

  toJson(): Record<string, unknown> {
    return {
      type: this.type,
      destination: this.destination,
      issuerId: this.issuerId,
      renterName: this.renterName,
      outDate: this.outDate.toISOString().split('T')[0],
      expectedReturnDate: this.expectedReturnDate.toISOString().split('T')[0],
    };
  }
}
